### PR TITLE
Add connector

### PR DESCRIPTION
**Pull request by stwa**
A connector is a helper to simplify building signal circuits by doing the following things:

* give access to signals by simple identifiers
* connect to specific signals without the need of explicit interconnections in code (loose coupling)
* take care of signal creation and lifetime management

### CODE REVIEW

File: src/internal/registry.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+const registry = new Map();
+
+/**
+ * Returns the handler for `kind` with `ìd`. If there is no handler associated
+ * with `kind` and `id`, returns `undefined`.
```
Comment (olle) [inline src/internal/registry.js]: For some reason "returns `undefined`" got me wondering about the intent and a possible difference between a handler _miss_ instead being a defined `null` for example. I have no idea if this means something.

File: src/internal/registry.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+ * Clears handlers from the registry. If `id` is undefined clears all handlers
+ * for `kind`. If `id` and `kind` is undefined, clears all handlers.
+ */
+export function clearHandlers(kind, id) {
+  if (kind === undefined && id === undefined) {
+    registry.clear();
+  } else if (id === undefined) {
+    registry.delete(kind);
+  } else {
+    registry.get(kind).delete(id);
```
Comment (olle) [inline src/internal/registry.js]: Is this call always safe, with regards to the chained call into the map for `kind`?

File: src/signal.js [hunk 3]
```diff
@@ -1,3 +1,16 @@
+/**
+ * Signals
```
Comment (olle) [inline src/signal.js:2]: Glorious documentation, reads very nice!

File: src/internal/registry.js (hunk 2 above)
Comment (stwa) [reply src/internal/registry.js]: You're right, it's not... I'm still wondering if it's a good idea to have this *global* registry where connectors register handlers by kind. Maybe it's better to create a registry per kind and use this internally in the corresponding connector module. What do you think?

File: src/internal/registry.js (hunk 1 above)
Comment (stwa) [reply src/internal/registry.js]: Hmm, maybe `undefined` says that I haven't really thought about it... Also, `undefined` might hint that a handler is just not defined - in contrast to `null`, which must always be assigned explicitly. There are other oddities in the JS language, for example `typeof undefined` evaluates to `undefined` whereas `typeof null` evaluates to `object`. Despite this, I have no strong opinion.

File: src/signal.js (hunk 3 above)
Comment (stwa) [reply src/signal.js:2]: Glad you like it!

File: src/internal/registry.js (hunk 2 above)
Comment (olle) [reply src/internal/registry.js]: I think it's quite alright at the moment. There's of course cohesion to gain from a move, but I don't think it's that important right now.

_First make it work._

Comment (stwa) [reply src/internal/registry.js]: I simplified this by completely removing the global registry. Will re-add it as soon as there is a real use case.

**Pull request merged**